include/zephyr/bluetooth/audio/bap.h

```diff
@@ -2048,14 +2048,28 @@ int bt_bap_broadcast_sink_delete(struct bt_bap_broadcast_sink *sink);
 /** @} */ /* End of group bt_bap_broadcast_sink */
 
 /**
- * @brief Register the callbacks for the Basic Audio Profile Scan Delegator
+ * @brief Register the Basic Audio Profile Scan Delegator and BASS.
  *
+ * Register the scan deligator and Broadcast Audio Scan Service (BASS)
+ * dynamically at runtime.
+ *
+ * @param cb Pointer to the callback struct
  * Only one set of callbacks can be registered at any one time, and calling this function multiple
  * times will override any previously registered callbacks.
  *
- * @param cb Pointer to the callback struct
+ * @return int   Error value. 0 on success, errno on fail.
  */
-void bt_bap_scan_delegator_register_cb(struct bt_bap_scan_delegator_cb *cb);
+int bt_bap_scan_delegator_register(struct bt_bap_scan_delegator_cb *cb);
```

Comment:
Is there any future plans to add a register parameter to set e.g. the number of receive states?

Reply:
I agree. Better to do this over 2 PRs to keep it simpler